### DISCUSSION TITLE
Doing something other than showing the +error.svelte page from the hook

### DISCUSSION THREAD

**Discussion by Madd0g** [sveltejs/kit#7196] [Q&A]
I know I can return `{ message }` from `handleError()` in `hooks.server.js` and get a message to show up in the error page.

Is there a way from `handleError()` to do something other than show the error page?
I'm wondering how to do something like optionally return json in case of an error?

thanks

**Comment by teemingc** [comment]
https://stackblitz.com/edit/sveltejs-kit-template-default-6zfpyv?file=src/routes/+error.svelte

Probably not what you're looking for, but the best I could come up with.
Maybe this should be a feature by allowing `handleError()` to return `Response`?

**Reply by Madd0g** [reply]
thanks.

this feels too hacky, feels like something `handleError` itself should be able to do

**Reply by teemingc** [reply]
Agreed. Perhaps you can open an issue regarding this.
